frontend/awx/views/jobs/JobDetails.tsx

```diff
@@ -1,25 +1,119 @@
-import { Skeleton } from '@patternfly/react-core';
-import { useParams } from 'react-router-dom';
-import { PageDetails, PageDetailsFromColumns } from '../../../../framework';
+import { Label, LabelGroup, Skeleton } from '@patternfly/react-core';
+import { Link, useParams } from 'react-router-dom';
+import {
+  DateTimeCell,
+  PageDetail,
+  PageDetails,
+  PageDetailsFromColumns,
+  useGetPageUrl,
+  usePageNavigate,
+} from '../../../../framework';
 import { useJobsColumns } from './hooks/useJobsColumns';
 import { useGetJob } from './JobPage';
 import { PageDetailCodeEditor } from '../../../../framework/PageDetails/PageDetailCodeEditor';
 import { useTranslation } from 'react-i18next';
 import { EmptyStateNoData } from '../../../../framework/components/EmptyStateNoData';
+import { StatusCell } from '../../../common/Status';
+import { AwxRoute } from '../../main/AwxRoutes';
+import { LastModifiedPageDetail } from '../../../common/LastModifiedPageDetail';
+import { ExecutionEnvironmentDetail } from '../../../common/ExecutionEnvironmentPageDetail';
 
 export function JobDetails() {
   const { t } = useTranslation();
   const params = useParams<{ id: string; job_type: string; job_id: string }>();
   const { job } = useGetJob(params.job_id ? params.job_id : params.id, params.job_type);
   const columns = useJobsColumns();
-
+  const getPageUrl = useGetPageUrl();
+  const pageNavigate = usePageNavigate();
   if (params.job_id && !job)
     return <EmptyStateNoData title="" description={t('Workflow job deleted')} />;
   if (!job) return <Skeleton />;
-
   return (
     <PageDetails>
       <PageDetailsFromColumns columns={columns} item={job} />
+      {job.playbook && <PageDetail label={t('Playbook')}>{job.playbook}</PageDetail>}
```

Comment:
PageDetail has an `isEmpty` prop you can use for all these conditionals. This approach should de-clutter the JSX quite a bit:

```suggestion
      <PageDetail label={t('Playbook')} isEmpty={!job.playbook}>{job.playbook}</PageDetail>
```